astropy/cosmology/tests/conftest.py

```diff
@@ -6,20 +6,70 @@
 # IMPORTS
 
 # STDLIB
+import inspect
 import json
 import os
 
+# THIRD-PARTY
 import pytest
 
+# LOCAL
 import astropy.cosmology.units as cu
 import astropy.units as u
 from astropy.cosmology import core
 from astropy.cosmology.core import Cosmology
 from astropy.tests.helper import pickle_protocol
 
+
 ###############################################################################
 # FUNCTIONS
 
+def get_redshift_methods(cosmology, allow_private=True, allow_z2=True):
+    """Get redshift methods from a cosmology.
+
+    Parameters
+    ----------
+    cosmology : |Cosmology| class or instance
+
+    Returns
+    -------
+    set[str]
+    """
+    methods = set()
+    for n in dir(cosmology):
+        try:  # get method, some will error on ABCs
+            m = getattr(cosmology, n)
+        except NotImplementedError:
+            continue
+
+        # Add anything callable, optionally excluding private methods.
+        if callable(m) and not (not allow_private and n.startswith("_")):
```

Comment:
Maybe `and (not n.startswith('_') or allow_private)` -- somehow, the double not made it tricky to follow.

Reply:
That is an improvement, thanks.